runway/blueprints/staticsite/cleanup.py

```diff
@@ -0,0 +1,376 @@
+#!/usr/bin/env python
+"""Cleanup Stack.
+
+Responsible for cleaning up the remaining orphaned resources created
+by the primary stack.
+"""
+from __future__ import print_function
+
+import json
+import logging
+from typing import Any, Dict, Union  # pylint: disable=unused-import
+
+import awacs.awslambda
+import awacs.cloudformation
+import awacs.iam
+import awacs.logs
+
+from awacs.aws import Allow, PolicyDocument, Statement
+from awacs.helpers.trust import make_simple_assume_policy
+from troposphere import (AccountId, Join, Output, Partition, Region, Sub,
+                         awslambda, iam, stepfunctions)
+
+from runway.cfngin.context import Context  # pylint: disable=unused-import
+from .staticsite import StaticSite
+
+LOGGER = logging.getLogger('runway')
+
+
+class Cleanup(StaticSite):
+    """Cleanup Stack."""
+
+    CLEANUP_VARIABLES = {
+        'function_arns': {'type': list,
+                          'default': [],
+                          'description': 'List of function ARNs that need to '
+                                         'be removed.'}
+    }
+
+    VARIABLES = {}  # type: Dict[str, Dict[str, Any]]
+
+    def __init__(self,
```

Comment:
> I think they're unique enough here to warrant just being in this stack.

I may be misunderstanding what you're trying to convey here. The reason this is being handled like this is because they are unique to this stack. If I add them in as part of the main variables then I'd have to either copy over all the variables from StaticSite (which isn't necessary since we're inheriting) or feed all of the variables into each one of the child templates.

Reply:
Put a different way: this stack doesn't seem related enough to the StaticSite to make it worth being a subclass of StaticSite. My preference would be to just make it a subclass of Blueprint and add the `stack_name` parameter.

Reply:
Updated